apps/web/app/(app)/onboarding/components/Objective.tsx

```diff
@@ -35,6 +35,46 @@ const Objective: React.FC<ObjectiveProps> = ({ next, skip, formbricksResponseId,
   const [selectedChoice, setSelectedChoice] = useState<string | null>(null);
   const [isProfileUpdating, setIsProfileUpdating] = useState(false);
 
+  const fieldsetRef = useRef<HTMLFieldSetElement>(null);
+
+  useEffect(() => {
+    const handleKeydown = (event: KeyboardEvent) => {
```

Comment:
Currently `handleKeyDown` is being repeated in `Objective.tsx` as well as` Role.tsx`. Also it can be used in other parts of our app as well so please abstract its logic and you can create a `util.ts` file in `apps/web/app/(app)/onboarding` and add that logic there. 

Also try refactoring `handleKeyDown` if possible  😊